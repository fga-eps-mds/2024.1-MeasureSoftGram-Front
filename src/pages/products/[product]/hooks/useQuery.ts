import { useEffect, useState } from 'react';

import { useRouter } from 'next/router';

import { RepositoriesTsqmiHistory } from '@customTypes/product';
import { productQuery } from '@services/product';

import { useProductContext } from '@contexts/ProductProvider';
import { useRepositoryContext } from '@contexts/RepositoryProvider';

import { getPathId } from '@utils/pathDestructer';

export const useQuery = () => {
  const { setCurrentProduct } = useProductContext();
  const { setRepositoryList, setRepositoriesLatestTsqmi } = useRepositoryContext();
  const [repositoriesTsqmiHistory, setRepositoriesTsqmiHistory] = useState<RepositoriesTsqmiHistory>();

  const { query } = useRouter();

  async function loadProduct(organizationId: string, productId: string) {
    try {
      const result = await productQuery.getProductById(organizationId, productId);
      setCurrentProduct(result.value);
    } catch (error) {
      // eslint-disable-next-line no-console
    }
  }

  async function loadRepositoriesTsqmiHistory(organizationId: string, productId: string) {
    try {
      const result = await productQuery.getProductRepositoriesTsqmiHistory(organizationId, productId as string);
      setRepositoriesTsqmiHistory(result.data);
    } catch (error) {
      // eslint-disable-next-line no-console
    }
  }

  async function loadRepositoriesLatestTsqmi(organizationId: string, productId: string) {
    try {
      const result = await productQuery.getProductRepositoriesLatestTsqmi(organizationId, productId as string);
      console.log(result.data);
      setRepositoriesLatestTsqmi(result.data);
    } catch (error) {
      // eslint-disable-next-line no-console
    }
  }

  async function loadRepositories(organizationId: string, productId: string) {
    try {
      const result = await productQuery.getAllRepositories(organizationId, productId as string);
      setRepositoryList(result.data.results);
    } catch (error) {
      // eslint-disable-next-line no-console
    }
  }

  useEffect(() => {
    if (query?.product) {
      const [organizationId, productId] = getPathId(query?.product as string);

      loadProduct(organizationId, productId);
      loadRepositoriesTsqmiHistory(organizationId, productId);
      loadRepositoriesLatestTsqmi(organizationId, productId);
      loadRepositories(organizationId, productId);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [query?.product]);

  return { repositoriesTsqmiHistory };
};
