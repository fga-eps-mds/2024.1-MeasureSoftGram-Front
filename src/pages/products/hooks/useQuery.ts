import { productQuery } from '@services/product';
import { useOrganizationContext } from '@contexts/OrganizationProvider';
import { useProductContext } from '@contexts/ProductProvider';
import { toast } from 'react-toastify';
import { useEffect } from 'react';

export const useQuery = () => {
  const { currentOrganization } = useOrganizationContext();
  const { updateProductList } = useProductContext();

  const loadAllProducts = async () => {
    if (currentOrganization) {
      try {
        const result = await productQuery.getAllProducts(currentOrganization.id);

        updateProductList(result.data.results);
      } catch (error) {
        toast.error(`${error}`);
      }
    }
  };

  useEffect(() => {
    if (currentOrganization) {
      loadAllProducts();
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [currentOrganization]);
};
