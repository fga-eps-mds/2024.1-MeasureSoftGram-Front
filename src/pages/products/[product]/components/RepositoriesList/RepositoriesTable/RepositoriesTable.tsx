import React, { useState, useEffect } from 'react';
import { useRouter } from 'next/router';
import { toast } from 'react-toastify';

import { TableContainer, Table, TableCell, TableHead, TableRow, TableBody, IconButton, Box } from '@mui/material';

import { styled } from '@mui/material/styles';
import { useRepositoryContext } from '@contexts/RepositoryProvider';
import { useProductContext } from '@contexts/ProductProvider';
import { useOrganizationContext } from '@contexts/OrganizationProvider';
import { Repository } from '@customTypes/repository';
import SearchButton from '@components/SearchButton';
import DeleteIcon from '@mui/icons-material/Delete';
import EditIcon from '@mui/icons-material/Edit';
import ConfirmationModal from '@components/ConfirmationModal';
import { FaGithub, FaGitlab, FaBitbucket, FaAws, FaCodeBranch } from 'react-icons/fa';
import { SiSubversion, SiMercurial, SiMicrosoftazure } from "react-icons/si";
import TsqmiBadge from '@pages/products/[product]/repositories/[repository]/components/TsqmiBadge';
import { useTranslation } from 'react-i18next';
import { useQuery } from '@hooks/useQuery';
import { productQuery } from '@services/product';
import { getPathId } from '@utils/pathDestructer';

interface Props {
  maxCount?: number;
}

const HoverIcon = styled('span')(({ theme }) => ({
  cursor: 'pointer',
  transition: 'opacity 0.3s',
  '&:hover': {
    opacity: 0.7,
  },
}));

const platformIcons = {
  'github': () => <FaGithub size="1.5em" />,
  'gitlab': () => <FaGitlab size="1.5em" />,
  'bitbucket': () => <FaBitbucket size="1.5em" />,
  'subversion (SVN)': () => <SiSubversion size="1.5em" />,
  'mercurial': () => <SiMercurial size="1.5em" />,
  'aws code commit': () => <FaAws size="1.5em" />,
  'azure repos': () => <SiMicrosoftazure size="1.5em" />,
  'outros': () => <FaCodeBranch size="1.5em" />,
};

const RepositoriesTable: React.FC<Props> = ({ maxCount }: Props) => {
  const { currentProduct } = useProductContext();
  const { currentOrganization, } = useOrganizationContext();
  const { repositoriesLatestTsqmi } = useRepositoryContext();
  const router = useRouter();
  const { handleRepositoryAction, loadProduct } = useQuery();

  const [filteredRepositories, setFilteredRepositories] = useState<Repository[]>([]);
  const [repositoryToDelete, setRepositoryToDelete] = useState<Repository | null>(null);
  const [showDeleteModal, setShowDeleteModal] = useState(false);
  const [repositoryList, setRepositoryList] = useState<Repository[]>([])
  const [confirmationName, setConfirmationName] = useState('');
  const [errorText, setErrorText] = useState('');


  useEffect(() => {
    const fetchData = async () => {
      try {
        const [organizationId, productId] = getPathId(router.query?.product as string);

        if (!currentProduct) {
          await loadProduct(organizationId, productId);
        }

        const result = await productQuery.getAllRepositories(organizationId, productId);
        setRepositoryList(result.data.results);
      } catch (error) {
        console.error(error);
        return [];
      }
    };

    fetchData();
  }, []); // The empty dependency array ensures this effect runs only once after the initial render


  const handleClickRedirects = (id: string) => {
    const path = `/products/${currentOrganization?.id}-${currentProduct?.id}-${currentProduct?.name}/repositories/${id}`;
    void router.push(path);
  };

  const getTsqmiValue = (id: number) => repositoriesLatestTsqmi?.results.find(result => result.id === id)?.current_tsqmi

  const getTsqmiUrl = (id: number) => (
    `${repositoriesLatestTsqmi?.results.find(result => result.id === id)?.url}badge`
  )

  function handleRepositoriesFilter(name: string) {
    if ((name == null || name === '') && repositoryList?.length) {
      setFilteredRepositories([...repositoryList]);
      return;
    }
    const repositoriesWithName =
      repositoryList?.filter((currentRepository: Repository) =>
        currentRepository.name.toLowerCase().includes(name.toLowerCase())
      ) ?? [];

    setFilteredRepositories([...repositoriesWithName]);
  }

  const openDeleteModal = (repository: Repository) => {
    setRepositoryToDelete(repository);
    setShowDeleteModal(true);
    setErrorText('')
  };

  const closeDeleteModal = () => {
    setShowDeleteModal(false);
    setConfirmationName('');
  };

  const confirmDelete = async () => {
    if (repositoryToDelete && currentOrganization && currentProduct) {
      try {
        const result = await handleRepositoryAction('delete', currentOrganization.id, currentProduct.id, String(repositoryToDelete.id), null);

        if (result.type === 'success') {
          const updatedRepositoryList = repositoryList?.filter((repo) => repo.id !== repositoryToDelete.id);
          setRepositoryList(updatedRepositoryList);
          toast.success(t('delete.toast-sucess'));
        } else {
          toast.error(t('delete.toast-error'));
        }
      } catch (error) {
        console.error('Error deleting repository:', error instanceof Error ? error.message : error);
      }
    }

    setShowDeleteModal(false);
  };

  useEffect(() => {
    if (repositoryList?.length) {
      setFilteredRepositories((prevState) => {
        const tempRepositoryList = [...repositoryList];

        const prevString = JSON.stringify(prevState);
        const currentString = JSON.stringify(tempRepositoryList);

        if (prevString !== currentString) {
          return tempRepositoryList;
        }

        return prevState;
      });
    }
  }, [repositoryList]);

  const { t } = useTranslation('repositories');

  return (
    <TableContainer>
      <Table>
        <TableHead>
          <TableRow>
            <TableCell style={{ paddingBottom: '35px' }}>Nome</TableCell>
            <TableCell align="right" style={{ paddingBottom: '35px' }}>
              <SearchButton
                onInput={(e) => handleRepositoriesFilter(e.target.value)}
                label={t('input-placeholder')}
              />
            </TableCell>
            <TableCell style={{ paddingBottom: '35px' }} />
          </TableRow>
        </TableHead>
        <TableBody>
          {filteredRepositories?.slice(0, maxCount ?? filteredRepositories.length).map((repo) => (
            <TableRow hover style={{ cursor: 'pointer' }} data-testid="repository-row" key={repo.id}>
              <TableCell onClick={() => handleClickRedirects(`${repo.id}-${repo.name}`)}>
                <Box display="flex" alignItems="center">
                  {repo.url ? (
                    <a href={repo.url} target="_blank" rel="noopener noreferrer" onClick={(e) => e.stopPropagation()}>
                      <HoverIcon>
                        {platformIcons[repo.platform] ? platformIcons[repo.platform]() : platformIcons.outros()}
                      </HoverIcon>
                    </a>
                  ) : (
                    <HoverIcon>
                      {platformIcons[repo.platform] ? platformIcons[repo.platform]() : platformIcons.outros()}
                    </HoverIcon>
                  )}
                  <span data-testid="repo-name" style={{ marginLeft: 10 }}>{repo.name}</span>
                </Box>
              </TableCell>
              <TableCell>
                <TsqmiBadge
                  latestTSQMI={getTsqmiValue(repo.id)}
                  latestTSQMIBadgeUrl={getTsqmiUrl(repo.id)}
                />
              </TableCell>
              <TableCell align="right">
                <IconButton aria-label="edit" onClick={() => router.push(`/products/${currentOrganization?.id}-${currentProduct?.id}/repositories/manage-repository?id=${repo.id}`)}>
                  <EditIcon />
                </IconButton>
                <IconButton aria-label="delete" onClick={() => openDeleteModal(repo)}>
                  <DeleteIcon />
                </IconButton>
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
      <ConfirmationModal
        open={showDeleteModal}
        onClose={closeDeleteModal}
        itemName={repositoryToDelete?.name ?? ''}
        onConfirm={confirmDelete}
        confirmationName={confirmationName}
        setConfirmationName={setConfirmationName}
        errorText={errorText}
        setErrorText={setErrorText}
      />
    </TableContainer>
  );
};

export default RepositoriesTable;
