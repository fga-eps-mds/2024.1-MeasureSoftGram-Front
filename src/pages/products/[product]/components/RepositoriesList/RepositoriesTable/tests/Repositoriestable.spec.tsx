import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import { useRouter } from 'next/router';
import { ProductProvider, useProductContext } from '@contexts/ProductProvider';
import { useOrganizationContext } from '@contexts/OrganizationProvider';
import { useRepositoryContext } from '@contexts/RepositoryProvider';
import { useQuery } from '@hooks/useQuery';
import { productQuery } from '@services/product';
import { act } from 'react-dom/test-utils';
import RepositoriesTable from '../RepositoriesTable'; // Replace with your actual path

// Mocking useRouter
jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}));

// Mocking contexts
jest.mock('@contexts/ProductProvider', () => ({
  useProductContext: jest.fn(),
}));

jest.mock('@contexts/OrganizationProvider', () => ({
  useOrganizationContext: jest.fn(),
}));

jest.mock('@contexts/RepositoryProvider', () => ({
  useRepositoryContext: jest.fn(),
}));

// Mocking custom hooks
jest.mock('@hooks/useQuery', () => ({
  useQuery: jest.fn(),
}));

// Mocking services
jest.mock('@services/product', () => ({
  productQuery: {
    getAllRepositories: jest.fn(),
  },
}));

// Mocking toast notifications
jest.mock('react-toastify', () => ({
  toast: {
    success: jest.fn(),
    error: jest.fn(),
  },
}));

describe('RepositoriesTable', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    useRouter.mockReturnValue({ push: jest.fn() });
    useProductContext.mockReturnValue({ currentProduct: { id: '1', name: 'Product A' } });
    useOrganizationContext.mockReturnValue({ currentOrganization: { id: '1' } });
    useRepositoryContext.mockReturnValue({ repositoriesLatestTsqmi: { results: [] } });
    useQuery.mockReturnValue({ handleRepositoryAction: jest.fn() });
  });

  it('renders repository list correctly', async () => {
    const mockRepositories = { data: { results: [{ id: 1, name: 'Repository 1', platform: 'github' }] } };
    productQuery.getAllRepositories.mockResolvedValue(mockRepositories);

    const { debug } = render(<RepositoriesTable />)
    debug();
    // Wait for the data fetching to complete
    await waitFor(() => {
      expect(screen.getByText('Repository 1')).toBeInTheDocument();
    });

    expect(productQuery.getAllRepositories).toHaveBeenCalledWith('1', '1'); // Assuming IDs for org and product
  });

  it('handles search input and filters repositories', async () => {
    const mockRepositories = { data: { results: [{ id: 1, name: 'Repository 1', platform: 'github' }] } };
    productQuery.getAllRepositories.mockResolvedValue(mockRepositories);

    render(<RepositoriesTable />);

    await waitFor(() => {
      expect(screen.getByText('Repository 1')).toBeInTheDocument();
    });

    act(() => {
      const searchInput = document.getElementById('search-bar');
      fireEvent.change(searchInput, { target: { value: 'Repo' } });
    })

    expect(screen.getByText('Repository 1')).toBeInTheDocument();
  });
});
