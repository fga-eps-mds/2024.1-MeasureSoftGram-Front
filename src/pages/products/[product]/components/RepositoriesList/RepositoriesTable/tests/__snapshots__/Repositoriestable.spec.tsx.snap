// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoriesTable Snapshot Deve corresponder ao Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
      >
        <table
          class="MuiTable-root css-rqglhn-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
                style="padding-bottom: 35px;"
              >
                Nome
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
                scope="col"
                style="padding-bottom: 35px;"
              >
                <div
                  aria-label="input-placeholder"
                  class="MuiFormControl-root MuiTextField-root text css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for="search-bar"
                    id="search-bar-label"
                  >
                    input-placeholder
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
                      id="search-bar"
                      placeholder="Buscar..."
                      type="text"
                      value=""
                    />
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-ittuaa-MuiInputAdornment-root"
                    >
                      <span
                        class="notranslate"
                      >
                        â€‹
                      </span>
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="SearchIcon"
                        focusable="false"
                        style="fill: #113d4c;"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                        />
                      </svg>
                    </div>
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-yjsfm1"
                      >
                        <span>
                          input-placeholder
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
                style="padding-bottom: 35px;"
              />
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
              data-testid="repository-row"
              style="cursor: pointer;"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <div
                  class="MuiBox-root css-70qvj9"
                >
                  <span
                    class="css-2w3t6l"
                  >
                    <svg
                      fill="currentColor"
                      height="1.5em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 384 512"
                      width="1.5em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M384 144c0-44.2-35.8-80-80-80s-80 35.8-80 80c0 36.4 24.3 67.1 57.5 76.8-.6 16.1-4.2 28.5-11 36.9-15.4 19.2-49.3 22.4-85.2 25.7-28.2 2.6-57.4 5.4-81.3 16.9v-144c32.5-10.2 56-40.5 56-76.3 0-44.2-35.8-80-80-80S0 35.8 0 80c0 35.8 23.5 66.1 56 76.3v199.3C23.5 365.9 0 396.2 0 432c0 44.2 35.8 80 80 80s80-35.8 80-80c0-34-21.2-63.1-51.2-74.6 3.1-5.2 7.8-9.8 14.9-13.4 16.2-8.2 40.4-10.4 66.1-12.8 42.2-3.9 90-8.4 118.2-43.4 14-17.4 21.1-39.8 21.6-67.9 31.6-10.8 54.4-40.7 54.4-75.9zM80 64c8.8 0 16 7.2 16 16s-7.2 16-16 16-16-7.2-16-16 7.2-16 16-16zm0 384c-8.8 0-16-7.2-16-16s7.2-16 16-16 16 7.2 16 16-7.2 16-16 16zm224-320c8.8 0 16 7.2 16 16s-7.2 16-16 16-16-7.2-16-16 7.2-16 16-16z"
                      />
                    </svg>
                  </span>
                  <span
                    data-testid="repo-name"
                    style="margin-left: 10px;"
                  >
                    2022-1-MeasureSoftGram-Core
                  </span>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <div />
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
              >
                <button
                  aria-label="edit"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="EditIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-label="delete"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="DeleteIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
    >
      <table
        class="MuiTable-root css-rqglhn-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
              style="padding-bottom: 35px;"
            >
              Nome
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
              scope="col"
              style="padding-bottom: 35px;"
            >
              <div
                aria-label="input-placeholder"
                class="MuiFormControl-root MuiTextField-root text css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="search-bar"
                  id="search-bar-label"
                >
                  input-placeholder
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
                    id="search-bar"
                    placeholder="Buscar..."
                    type="text"
                    value=""
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-ittuaa-MuiInputAdornment-root"
                  >
                    <span
                      class="notranslate"
                    >
                      â€‹
                    </span>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="SearchIcon"
                      focusable="false"
                      style="fill: #113d4c;"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                      />
                    </svg>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        input-placeholder
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
              style="padding-bottom: 35px;"
            />
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
            data-testid="repository-row"
            style="cursor: pointer;"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              <div
                class="MuiBox-root css-70qvj9"
              >
                <span
                  class="css-2w3t6l"
                >
                  <svg
                    fill="currentColor"
                    height="1.5em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 384 512"
                    width="1.5em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M384 144c0-44.2-35.8-80-80-80s-80 35.8-80 80c0 36.4 24.3 67.1 57.5 76.8-.6 16.1-4.2 28.5-11 36.9-15.4 19.2-49.3 22.4-85.2 25.7-28.2 2.6-57.4 5.4-81.3 16.9v-144c32.5-10.2 56-40.5 56-76.3 0-44.2-35.8-80-80-80S0 35.8 0 80c0 35.8 23.5 66.1 56 76.3v199.3C23.5 365.9 0 396.2 0 432c0 44.2 35.8 80 80 80s80-35.8 80-80c0-34-21.2-63.1-51.2-74.6 3.1-5.2 7.8-9.8 14.9-13.4 16.2-8.2 40.4-10.4 66.1-12.8 42.2-3.9 90-8.4 118.2-43.4 14-17.4 21.1-39.8 21.6-67.9 31.6-10.8 54.4-40.7 54.4-75.9zM80 64c8.8 0 16 7.2 16 16s-7.2 16-16 16-16-7.2-16-16 7.2-16 16-16zm0 384c-8.8 0-16-7.2-16-16s7.2-16 16-16 16 7.2 16 16-7.2 16-16 16zm224-320c8.8 0 16 7.2 16 16s-7.2 16-16 16-16-7.2-16-16 7.2-16 16-16z"
                    />
                  </svg>
                </span>
                <span
                  data-testid="repo-name"
                  style="margin-left: 10px;"
                >
                  2022-1-MeasureSoftGram-Core
                </span>
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              <div />
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
            >
              <button
                aria-label="edit"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="EditIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-label="delete"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="DeleteIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
